docs/rules/no-try-expect.md

```diff
@@ -0,0 +1,56 @@
+# Prevent catch assertions in tests (no-try-expect)
+
+This rule prevents the use of `expect` inside `catch` blocks.
+
+## Rule Details
+
+Expectations inside a `catch` block can be silently skipped. While Jest provides
+an `expect.assertions(number)` helper, it might be cumbersome to add this to
```

Comment:
Wondering if the rule should check for these and pass if the test uses it, for the cases where one might actually need complicated logic in a catch

Reply:
in that case I think an `eslint-disable-next-line` comments is better rather than try to make the rule too clever.

Reply:
Not if you `eslint-disable` without adding `expect.assertions` - the lint error could suggest adding `expect.assertions` to get rid of the warning and save someone from mindlessly disabling it